src/transformStateWithClones.js

```diff
@@ -8,6 +8,44 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const transformedClones = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const clone = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          clone[data] = action.extraData[data];
+        }
```

Comment:
you can just use general object methods
```suggestion
        Object.assign(stateCopy, action.extraData);
```